From 61473a9529a931ad35ee83762967df2f5118610d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Sat, 11 Aug 2012 13:29:25 +0200
Subject: [PATCH 10/10] contrib/ebml+mkv: compile with hidden symbols

This solves linking issues on iOS and should have no negative impact on further platforms
---
 contrib/src/ebml/rules.mak     | 4 ++--
 contrib/src/matroska/rules.mak | 4 +++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/contrib/src/ebml/rules.mak b/contrib/src/ebml/rules.mak
index 9d87c29..4a5ac16 100644
--- a/contrib/src/ebml/rules.mak
+++ b/contrib/src/ebml/rules.mak
@@ -14,10 +14,10 @@ libebml: libebml-$(EBML_VERSION).tar.bz2 .sum-ebml
 	$(MOVE)
 
 # libebml requires exceptions
-EBML_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fexceptions" \
+EBML_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fexceptions -fvisibility=hidden" \
 					CPPFLAGS=""
 
 .ebml: libebml
-	cd $< && $(HOSTVARS) ./configure $(HOSTCONF)
+	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(EBML_EXTRA_FLAGS)
 	cd $< && $(MAKE) install
 	touch $@
diff --git a/contrib/src/matroska/rules.mak b/contrib/src/matroska/rules.mak
index b8d0165..c71ee83 100644
--- a/contrib/src/matroska/rules.mak
+++ b/contrib/src/matroska/rules.mak
@@ -16,7 +16,9 @@ libmatroska: libmatroska-$(MATROSKA_VERSION).tar.bz2 .sum-matroska
 	$(UNPACK)
 	$(MOVE)
 
+MATROSKA_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fvisibility=hidden"
+
 .matroska: libmatroska
-	cd $< && $(HOSTVARS) ./configure $(HOSTCONF)
+	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(MATROSKA_EXTRA_FLAGS)
 	cd $< && $(MAKE) install
 	touch $@
-- 
1.9.3 (Apple Git-50)

